### PR TITLE
Fix stab for TextIOBase.detach() and add stab for TextIOWrapper.buffer

### LINKED ISSUE #3993: The result of TextIOWrapper.detach() should be IO[bytes]
For the following example:
```python
import sys, io

if isinstance(sys.stdout, io.TextIOWrapper) and sys.version_info < (3, 7):
    sys.stdout = io.TextIOWrapper(
        sys.stdout.detach(),
        encoding="ascii",
        errors="replace")
```
I get the following error:
```
reconfigure36.py:5: error: Argument 1 to "TextIOWrapper" has incompatible type "IOBase"; expected "IO[bytes]"
```
Should not `TextIOWrapper.detach()` return `IO[bytes]` instead of `IOBase`?

Replacing `sys.stdout.detach()` with `sys.stdout.buffer` fixes the complains.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Closes #3993.

### CODE REVIEW

File: stdlib/3/io.pyi [hunk 1]
```diff
@@ -164,13 +164,15 @@ class TextIOWrapper(TextIO):
     #              -> None: ...
     def __init__(
         self,
-        buffer: IO[bytes],
+        buffer: BinaryIO,
```
Comment (srittau) [inline stdlib/3/io.pyi:167]: I believe CI fails due to this change. Honestly, I would just leave this instance as `IO[bytes]` for now. In the future it might make sense to make `TextIOWrapper` generic over the buffer, but I haven't looked at it in depth.

**Comment by hauntsaninja** [comment]
As it happens, I just removed that particular sample from mypy (https://github.com/python/mypy/pull/8838), so if you re-run CI it should pass.

**Comment by srittau** [comment]
I'm re-running it. It would be great if someone with a large typed code base (hey Dropbox team) could check this, though. Our internal code base doesn't use `TextIOWrapper` directly.
